lib/setupHooks/inheritCargoArtifactsHook.sh

```diff
@@ -17,10 +17,18 @@ inheritCargoArtifacts() {
 
   mkdir -p "${cargoTargetDir}"
   if [ -f "${preparedArtifacts}" ]; then
+
+    if [ -f "${preparedArtifacts}.prev" ]; then
+      inheritCargoArtifacts "$(readlink -f "${preparedArtifacts}.prev")" "$cargoTargetDir"
+    fi
+
     echo "decompressing cargo artifacts from ${preparedArtifacts} to ${cargoTargetDir}"
 
     zstd -d "${preparedArtifacts}" --stdout | \
       tar -x -C "${cargoTargetDir}" --strip-components=1
+    rm -f "${cargoTargetDir}/.crane-previous-archive"
+    ln -s "${preparedArtifacts}" "${cargoTargetDir}/.crane-previous-archive"
```

Comment:
This works fine, though technically we're wasting a few operations waiting for the file system to delete/create the new marker file. Maybe we could store this in a (global) variable to be a bit cheaper?

Reply:
I did not know (still don't) how to do that. :D

Reply:
Maybe we can land as is and then follow-up with improvement?

BTW. I'm totally fine if you want to just redo/tweak and merge this PR as you please. For final touches it is generally faster than going back and forth.  :)

Reply:
No problem, I can take it to the finish line sometime this week!